### PR TITLE
[CHORE] rename Record to RecordInstance

### PR DESCRIPTION
**Pull request by runspired**
Replaces #6350

### CODE REVIEW

File: packages/store/addon/-private/ts-interfaces/ds-model.ts [hunk 1]
```diff
@@ -1,11 +1,10 @@
-import { Record } from './record';
 import RSVP from 'rsvp';
 import EmberObject from '@ember/object';
 import { JsonApiValidationError } from './record-data-json-api';
 import { RelationshipSchema } from './record-data-schemas';
 
 // Placeholder until model.js is typed
-export interface DSModel extends Record, EmberObject {
+export interface DSModel extends EmberObject {
```
Comment (igorT) [inline packages/store/addon/-private/ts-interfaces/ds-model.ts]: Missing RecordInstance here

File: packages/store/addon/-private/ts-interfaces/record.ts [hunk 2]
```diff
@@ -10,8 +10,7 @@
   model instances (aka `Records`) are completely user supplied and opaque to the internals, we need a type
   through which to communicate what is valid.
 
-  The type belows allows for either a class instance, or an object, but not primitive values or functions.
+  The type belows allows for anything extending object.
 */
 
-// TODO Rename to RecordInstance
-export type Record = Object;
+export type RecordInstance = object;
```
Comment (igorT) [inline packages/store/addon/-private/ts-interfaces/record.ts]: What is the difference between lowercase and capitalized object?

File: packages/store/addon/-private/ts-interfaces/record.ts [hunk 3]
```diff
@@ -10,8 +10,7 @@
   model instances (aka `Records`) are completely user supplied and opaque to the internals, we need a type
   through which to communicate what is valid.
 
-  The type belows allows for either a class instance, or an object, but not primitive values or functions.
+  The type belows allows for anything extending object.
```
Comment (igorT) [inline packages/store/addon/-private/ts-interfaces/record.ts]: We should rename this file now that the type has been renamed

File: packages/store/addon/-private/ts-interfaces/record.ts (hunk 2 above)
Comment (igorT) [inline packages/store/addon/-private/ts-interfaces/record.ts]: Should we make this the default export?

**Comment by igorT** [comment]
You mentioned this replaces #6350, are there substantial changes, or is it just a rebase?

**Comment by igorT** [comment]
cc @rwjblue

File: packages/store/addon/-private/ts-interfaces/record.ts (hunk 2 above)
Comment (runspired) [reply packages/store/addon/-private/ts-interfaces/record.ts]: ack, thanks, that was an accidental change

File: packages/store/addon/-private/ts-interfaces/ds-model.ts (hunk 1 above)
Comment (runspired) [reply packages/store/addon/-private/ts-interfaces/ds-model.ts]: it's intentional, it should not be extending `RecordInstance`

File: packages/store/addon/-private/ts-interfaces/record.ts (hunk 2 above)
Comment (runspired) [reply packages/store/addon/-private/ts-interfaces/record.ts]: I'd rather not as there seems to be an annoying thing where default exports warn during transpile when they were just a type

**Comment by runspired** [comment]
@igorT it's a re-create since files have been re-arranged since

**Pull request merged**